whisper.cpp

```diff
@@ -3299,20 +3357,26 @@ static std::string whisper_openvino_get_path_cache(std::string path_bin) {
 struct whisper_state * whisper_init_state(whisper_context * ctx) {
     whisper_state * state = new whisper_state;
 
-    state->backend = whisper_backend_init(ctx->params);
-    if (!state->backend) {
+    state->backends = whisper_backend_init(ctx->params);
+    if (state->backends.empty()) {
         WHISPER_LOG_ERROR("%s: whisper_backend_init() failed\n", __func__);
         whisper_free_state(state);
         return nullptr;
     }
 
-    state->mel_calc = whisper_mel_calc_create(state->backend, ctx->model.filters);
+    state->backends_used.push_back(ctx->backend);
+    state->backends_used.insert(state->backends_used.end(), state->backends.begin(), state->backends.end());
+    for (int i = 0; i < (int) state->backends_used.size(); ++i) {
+        printf("state->backends_used[%d] = %p, name = %s\n", i, state->backends_used[i], ggml_backend_name(state->backends_used[i]));
+    }
```

Comment:
The backend instance in `whisper_context` is used to create a read-only buffer that contains all tensors of the model. Do I need to pass this backend together with `whisper_state` backends when initiating the schedulers? Currently it works both ways

Reply:
I think this is actually not necessary and can have side-effect of not using the BLAS backend if the `whisper_context` backend is CPU. So I'll pass just the `whisper_state` backends to the schedulers

Reply:
Backend instances are not necessary at all to create buffers at all, it's just more convenient because you can use the `ggml_backend_get_default_buffer_type` to obtain the preferred buffer type of this backend, rather than having to call a backend-specific function such as `ggml_backend_metal_buffer_type`, which would require a bunch of `#ifdefs` such as in `llama_default_buffer_type_offload`. But the backend instance is not bound to the buffer in any way, and it is not necessary to pass it to the sched.

Reply:
Got it, will try to remove the `backend` instance from `whisper_context` then to simplify things